### ISSUE TITLE
Classifying songs

### ISSUE THREAD

**Issue by Phyks** [Polochon-street/bliss#29]
Hi,

I've just thought again about the k-means quick analysis you performed. Did you manage to somehow form "clusters" according to music genres?

Typically, do you think it would be possible to form a "rock" cluster, a "classical" cluster and so on to try to guess the genre of an unknown music?

My idea behind it is, besides the obvious thematic radio _à la_ Spotify, to be able to make a playlist drift from one genre to another one, either because I am listening to pop, but want the music to drift toward jazz as time passes and as I will be going to bed, or to use it for some "education"/discovery pupose, starting from a music the user likes and knows well, and going toward another genre in a continuous manner.

**Comment by Polochon-street** [comment]
Hi,

Well as seen in the picture, the more genres you add to the clustering, the less Bliss is able to perform clustering successfully:
![Bliss vs random graph](https://img.linuxfr.org/img/687474703a2f2f692e696d6775722e636f6d2f444239364c634d2e706e67/DB96LcM.png "Bliss vs random graph")
Because I used very dissimilar genres, I'd say roughly half of the tracks are distributed into right clusters, and half are just scattered around genres that are similar, but not exactly the same (some blues songs in jazz, etc).

So, in short, I think it would still be possible to make « genre » clouds based on Bliss' output values; it would just need a huge amount of data. I can clean and upload the Octave file I used to make the graph if you want to fiddle with it.

**Comment by Phyks** [comment]
Cool! :)
